homeassistant/components/recorder/__init__.py

```diff
@@ -255,8 +191,10 @@ def run(self):
                 self._close_connection()
                 self.queue.task_done()
                 return
-
-            if event.event_type == EVENT_TIME_CHANGED:
+            elif event is purge_task:
+                purge.purge_old_data(self, self.purge_days)
```

Comment:
:+1: probably never got issues since this happens so infrequently, but much better as part of this worker loop